Objects/tupleobject.c

```diff
@@ -495,52 +495,47 @@ tupleconcat(PyTupleObject *a, PyObject *bb)
 static PyObject *
 tuplerepeat(PyTupleObject *a, Py_ssize_t n)
 {
-    Py_ssize_t size;
-    PyTupleObject *np;
-    if (Py_SIZE(a) == 0 || n == 1) {
+    const Py_ssize_t input_size = Py_SIZE(a);
+    if (input_size == 0 || n == 1) {
         if (PyTuple_CheckExact(a)) {
             /* Since tuples are immutable, we can return a shared
                copy in this case */
             Py_INCREF(a);
             return (PyObject *)a;
         }
     }
-    if (Py_SIZE(a) == 0 || n <= 0) {
+    if (input_size == 0 || n <= 0) {
         return tuple_get_empty();
     }
-    if (n > PY_SSIZE_T_MAX / Py_SIZE(a))
+    assert(n>0);
+    assert(input_size > 0);
```

Comment:
Can we copy/move these assertions to inside `_Py_memory_repeat`?

Reply:
The `assert(input_size > 0);` is indeed better in the `_Py_memory_repeat`. The `assert(n>0)` is there for the line with `input_size > PY_SSIZE_T_MAX / n`. We could remove it here, but I see no need to add it to the `_Py_memory_repeat`